lib/hci-socket/smp.js

```diff
@@ -28,14 +29,21 @@ var Smp = function(aclStream, localAddressType, localAddress, remoteAddressType,
 
   this._aclStream.on('data', this.onAclStreamDataBinded);
   this._aclStream.on('end', this.onAclStreamEndBinded);
+
+  this._remoteAddress = remoteAddress;
+
+  if(passkeyProvider)
+    this._passkeyProvider = passkeyProvider;
+  else
+    this._passkeyProvider = Promise.resolve;
```

Comment:
Line 38 should be `this._passkeyProvider = Promise.resolve()` right?

Reply:
Good catch.  Also very odd.  What kind of sense does this make anyway?
```
> var mockProvider = Promise.resolve
undefined
> mockProvider
[Function: resolve]
> mockProvider()
TypeError: mockProvider is not a function
    at resolve (native)
    at repl:1:1
    at REPLServer.defaultEval (repl.js:248:27)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.<anonymous> (repl.js:412:12)
    at emitOne (events.js:82:20)
    at REPLServer.emit (events.js:169:7)
    at REPLServer.Interface._onLine (readline.js:210:10)
    at REPLServer.Interface._line (readline.js:549:8)
> mockProvider = function resolve() {}
[Function: resolve]
> mockProvider
[Function: resolve]
> mockProvider()
undefined
```

Reply:
I did actually intend to set it to the function reference, but that appears not to work.